### PR TITLE
Fixed linking on PS4

### PR DESCRIPTION
**Pull request by georged-youi**

### CODE REVIEW

File: xmlmemory.c [hunk 1]
```diff
@@ -978,13 +978,13 @@ xmlInitMemory(void)
      xmlMemInitialized = 1;
      xmlMemMutex = xmlNewMutex();
 
-#ifdef HAVE_STDLIB_H
+#ifdef defined(HAVE_STDLIB_H) && !defined(__ORBIS__)
```
Comment (Halmo) [inline xmlmemory.c]: Could just include YiPort.h there is a getenv in there.

Comment (georged-youi) [reply xmlmemory.c]: `YiPort.h` is not available, as this is build as a standalone library

File: catalog.c [hunk 2]
```diff
@@ -3091,8 +3091,10 @@ xmlInitializeCatalogData(void) {
     if (xmlCatalogInitialized != 0)
 	return;
 
+#if !defined(__ORBIS__)
```
Comment (Halmo) [inline catalog.c]: Same comment about getenv as the other.

**Comment by mathieumallet** [comment]
not sure but I think generally the merge-into branch shouldn't have the version number (and instead we should be using tags for that)

**Review by mathieumallet — APPROVED** [review]

File: xmlmemory.c [hunk 3]
```diff
@@ -978,13 +978,13 @@ xmlInitMemory(void)
...
      breakpoint = getenv("XML_MEM_BREAKPOINT");
      if (breakpoint != NULL) {
          sscanf(breakpoint, "%ud", &xmlMemStopAtBlock);
      }
 #endif
-#ifdef HAVE_STDLIB_H
+#ifdef defined(HAVE_STDLIB_H) && !defined(__ORBIS__)
```
Comment (mathieumallet) [inline xmlmemory.c]: `#ifdef defined(`? that's probably a bug, should be `#if defined(...`

Comment (georged-youi) [reply xmlmemory.c]: yep, it's a bug.

File: xmlmemory.c (hunk 1 above)
Comment (georged-youi) [reply xmlmemory.c]: fixed

File: catalog.c (hunk 2 above)
Comment (georged-youi) [reply catalog.c]: fixed

File: xmlmemory.c (hunk 3 above)
Comment (georged-youi) [reply xmlmemory.c]: fixed

**Review by patrickyoui — APPROVED** [review]
Looks good

**Pull request merged**